src/System.Windows.Forms/src/System/Windows/Forms/Controls/TreeView/TreeNodeCollection.cs

```diff
@@ -219,7 +219,11 @@ public virtual void AddRange(params TreeNode[] nodes)
             tv.BeginUpdate();
         }
 
-        _owner.Nodes.FixedIndex = _owner._childNodes.Count;
+        if (tv is not null && !tv.Sorted)
```

Comment:
```suggestion
        if (tv is null || !tv.Sorted)
```

This is the reason of the test failures

Reply:
Let me correct it.